state/aerospike/aerospike.go

```diff
@@ -0,0 +1,243 @@
+	arr, err := json.Marshal(req.Value)
+	if err != nil {
+		return err
+	}
+	err = json.Unmarshal(arr, &data)
+	if err != nil {
+		return err
+	}
+	err = aspike.client.Put(writePolicy, asKey, as.BinMap(data))
+	if err != nil {
+		return fmt.Errorf("aerospike: failed to save value for key %s - %v", req.Key, err)
+	}
+	return nil
+}
+
+// BulkSet performs a bulks save operation
+func (aspike *Aerospike) BulkSet(req []state.SetRequest) error {
+	for _, s := range req {
+		err := aspike.Set(&s)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Get retrieves state from Aerospike with a key
+func (aspike *Aerospike) Get(req *state.GetRequest) (*state.GetResponse, error) {
+	asKey, err := as.NewKey(aspike.namespace, aspike.set, req.Key)
+
+	if err != nil {
+		return nil, err
+	}
+
+	policy := &as.BasePolicy{}
+	if req.Options.Consistency == state.Strong {
+		policy.ConsistencyLevel = as.CONSISTENCY_ALL
+	} else {
+		policy.ConsistencyLevel = as.CONSISTENCY_ONE
+	}
+	record, err := aspike.client.Get(policy, asKey)
+	if err != nil {
+		return nil, fmt.Errorf("aerospike: failed to get value for key %s - %v", req.Key, err)
+	}
+	value, err := aspike.json.Marshal(record.Bins)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &state.GetResponse{
+		Data: value,
+		ETag: fmt.Sprintf("%d", record.Generation),
+	}, nil
+}
+
+// Delete performs a delete operation
+func (aspike *Aerospike) Delete(req *state.DeleteRequest) error {
+	err := state.CheckDeleteRequestOptions(req)
+	if err != nil {
+		return err
+	}
+	writePolicy := &as.WritePolicy{}
+
+	if req.ETag != "" {
+		var gen uint32
+		gen, err = convertETag(req.ETag)
+		if err != nil {
+			return errInvalidETag
+		}
+		// pass etag and fail writes is etag in DB is not same as passed by dapr (EXPECT_GEN_EQUAL)
+		writePolicy.Generation = gen
+		writePolicy.GenerationPolicy = as.EXPECT_GEN_EQUAL
+	}
+
+	if req.Options.Consistency == state.Strong {
+		// COMMIT_ALL indicates the server should wait until successfully committing master and all replicas.
+		writePolicy.CommitLevel = as.COMMIT_ALL
+	} else {
+		writePolicy.CommitLevel = as.COMMIT_MASTER
+	}
+
+	asKey, err := as.NewKey(aspike.namespace, aspike.set, req.Key)
+	if err != nil {
+		return err
+	}
+
+	_, err = aspike.client.Delete(writePolicy, asKey)
+	if err != nil {
+		return fmt.Errorf("aerospike: failed to delete key %s - %v", req.Key, err)
+	}
+	return nil
+}
+
+// BulkDelete performs a bulk delete operation
+func (aspike *Aerospike) BulkDelete(req []state.DeleteRequest) error {
+	for _, re := range req {
+		err := aspike.Delete(&re)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func parseHosts(hostsMeta string) ([]*as.Host, error) {
+	hostPorts := []*as.Host{}
+	for _, hostPort := range strings.Split(hostsMeta, ",") {
+		if !strings.Contains(hostPort, ":") {
+			return nil, errInvalidHosts
+		}
+		host := strings.Split(hostPort, ":")[0]
+		port, _ := strconv.Atoi(strings.Split(hostPort, ":")[1])
```

Comment:
I would use strconv.ParseUInt instead of Atoi for the safety.

Reply:
done